src/lib.rs

```diff
@@ -606,25 +614,24 @@ impl<T: Counter> Histogram<T> {
         // NOT ok to be +/- 2 units at 1999. Only starting at 2000. So internally, we need to
         // maintain single unit resolution to 2x 10^decimal_points.
 
-        // largest value with single unit resolution
-        let largest = 2 * 10_u64.pow(sigfig);
+        // largest value with single unit resolution, in [2, 200_000].
+        let largest = 2 * 10_u32.pow(sigfig as u32);
 
-        let unit_magnitude = ((low as f64).log2() / 2_f64.log2()).floor() as usize;
```

Comment:
`log_2(2)` is 1, so dividing by it isn't needed. The Java code used division with the natural log to calculate `log_2`, but fortunately we need not resort to such things.